modules/programs/antidote.nix

```diff
@@ -0,0 +1,71 @@
+{ config, lib, pkgs, ... }:
+
+with builtins;
+with lib;
+let
+  module_config = config.programs.zsh.antidote;
+
+  zPluginStr = with lib;
+    (pluginNames:
+      optionalString (pluginNames != [ ]) "${concatStrings (map (name: ''
+        ${name}
+      '') pluginNames)}");
+  trimLines = (text:
+    concatLines (map (p: removePrefix "     " p) (splitString "\n" text)));
+in {
+  options.programs.zsh.antidote = {
+    enable = mkEnableOption "antidote - a zsh plugin manager";
+
+    plugins = mkOption {
+      default = [ ];
+      type = types.listOf types.str;
+      description = "List of antidote plugins, strings.";
+    };
+
+    useFriendlyNames = mkOption {
+      default = false;
+      type = types.bool;
+      description = "use friendlyPath.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.antidote;
+      description = "custom antidote pacakge.";
+    };
+
+    antidoteHome = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      defaultText = "~/.antidote or ~/Library/Caches/antidote(darwin).";
+      apply = toString;
+      description = "Path to antidote home directory, cache contents.";
+    };
```

Comment:
In which case would it be necessary to set a custom value here? My spontaneous thought would be to removing this option entirely.

Reply:
Cache path is not important to custom.  since it is a undocumented option, removed.